tests/rptest/tests/topic_recovery_test.py

```diff
@@ -683,6 +540,47 @@ def generate_baseline(self):
             producer.wait()
             producer.free()
 
+    def _collect_non_data_batch_sizes(self, expected_topics):
+        """
+        Parses segment file and calculates sizes of all non-data batches present in the file.
+        These sizes are used when comparing S3 segment sizes with restored data, to account
+        for differences caused because configuration batches are not written to restored data.
+        """
+        non_data_batches_per_ntp = defaultdict(lambda: 0)
+        segments = [
+            item for item in self._s3.list_objects(self._bucket)
+            if not item.Key.endswith('manifest.json')
```

Comment:
after filtering out all manifests you will have log segments and also transactional manifests (*.tx extension)
but here, if you don't use transactions no tx manifests will be created